addons/website/static/src/js/content/website_root.js

```diff
@@ -46,6 +46,23 @@ export const WebsiteRoot = publicRootData.PublicRoot.extend(KeyboardNavigationMi
         if (/^((?!chrome|android).)*safari/i.test(navigator.userAgent) && document.querySelector('#wrapwrap')) {
             document.querySelector('#wrapwrap').classList.add('o_safari_browser');
         }
+        // Simplify local links.
+        const location = new URL(window.location.href);
+        location.hash = "";
+        if (!location.pathname.endsWith("/")) {
+            location.pathname += "/";
+        }
+        for (const linkEl of document.querySelectorAll("#wrapwrap a[href*='#']:not([href$='#'])")) {
+            const url = new URL(linkEl.href);
+            const hash = url.hash;
+            url.hash = "";
+            if (!url.pathname.endsWith("/")) {
+                url.pathname += "/";
+            }
+            if (hash && location.toString() === url.toString()) {
+                linkEl.href = hash;
+            }
+        }
```

Comment:
@bso-odoo Not a fan of changing the URLs in javascript on page load. Given the fact that the scrolling animation is handled in JS anyway, can't you do this when clicking the anchor link only? (instead of on page load?)

Note: I am also kinda surprise it works, I though that the link `/#XXX` would not be transformed into `/fr#XXX` during page rendering... so this is the case in fact? For all links in the page?

Reply:
@qsm-odoo It's translated for more than just links: a bunch of attributes runs through `url_for` (which then runs through `url_lang`) <https://github.com/odoo/odoo/blob/17.0/addons/website/models/ir_qweb.py#L26~L32>

I have moved the `href` conversion into `scroller_service.js` - and conditioned it so that it does not get applied on `/web`. But this is not the class that makes the scroll happen on website pages - the `scrollTo` utility function is not even called during the transition.